x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/StartTrainedModelDeploymentAction.java

```diff
@@ -357,7 +357,7 @@ public ActionRequestValidationException validate() {
                 if (numberOfAllocations < 1) {
                     validationException.addValidationError("[" + NUMBER_OF_ALLOCATIONS + "] must be a positive integer");
                 }
-                if (adaptiveAllocationsSettings != null && adaptiveAllocationsSettings.getEnabled()) {
```

Comment:
Why do we need to add `== Boolean.TRUE`?

Reply:
`getEnabled()` returns a nullable Boolean object, which will throw a NPE when java tries to unbox it to the primitive for the if check